apps/api/src/server/app/inspector/inspector.validators.js

```diff
@@ -0,0 +1,94 @@
+// @ts-check
+
+import { composeMiddleware } from '@pins/express';
+import { validateFutureDate } from '@pins/platform';
+import { body, header, validationResult } from 'express-validator';
+import appealRepository from '../repositories/appeal.repository.js';
+
+/** @typedef {import('@pins/inspector').SiteVisitType} SiteVisitType */
+
+/** @type {import('express').RequestHandler } */
+export const validateUserId = async (request, response, next) => {
+	const result = await header('userId')
+		.notEmpty()
+		.bail()
+		.withMessage('Authentication error. Missing header `userId`.')
+		.toInt()
+		.run(request);
+
+	if (!result.isEmpty()) {
+		response.status(401).send({ errors: result.formatWith(({ msg }) => msg).mapped() });
```

Comment:
Not sure what your approach to error messaging is. I would have assumed that if we have an error handler, then all errors should go through it for consistency. Therefore, here I would throw an error

Reply:
yeah, we can do that. So you want to:

`throw new InspectorError({ errors: result.formatWith(({ msg }) => msg).mapped() }, 401);` 

right?